core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -580,24 +587,37 @@ private void closeAndNotify(DiscoveryNode node, NodeChannels nodeChannels, Strin
     /**
      * Disconnects from a node if a channel is found as part of that nodes channels.
      */
-    protected final void disconnectFromNodeChannel(final Channel channel, final Exception failure) {
+    protected final void disconnectFromNodeChannel(final Channel channel, final String reason) {
         threadPool.generic().execute(() -> {
             try {
-                try {
+                if (isOpen(channel)) {
```

Comment:
question - why did you add this? it seems all channels has double closing protection?